secure/net/src/lib.rs

```diff
@@ -15,14 +15,123 @@
 //! Internally both the client and server leverage a NetworkStream that communications in blocks
 //! where a block is a length prefixed array of bytes.
 
-use libra_logger::{debug, trace};
+use libra_logger::{info, trace, warn, Schema};
+use libra_secure_push_metrics::{register_int_counter_vec, IntCounterVec};
+use once_cell::sync::Lazy;
+use serde::Serialize;
 use std::{
     io::{Read, Write},
     net::{Shutdown, SocketAddr, TcpListener, TcpStream},
     thread, time,
 };
 use thiserror::Error;
 
+#[derive(Schema)]
+struct SecureNetLogSchema<'a> {
+    service: &'static str,
+    mode: NetworkMode,
+    event: LogEvent,
+    #[schema(debug)]
+    remote_peer: Option<&'a SocketAddr>,
```

Comment:
So, this is something that is not great about ES, is if you have two things that serialize to the same field in different ways, it could reject one of them.  As long as both `SocketAddr` and `PeerId` serialize to a single string it should be fine, but just a weird caveat.